components/search/RegionSearch/RegionSelect.tsx

```diff
@@ -78,6 +83,7 @@ const RegionSelect: React.FC<RegionSelectProps> = ({
 
   return (
     <div>
+      <LocationCombobox searchLocations={handleFindRegion} setRegionId={onRegionSelect} />
```

Comment:
```suggestion
```

Since this is `Region Search`. We only need `query`, `domain`, `locale`. All of this is available + more in `HomeSearchPage` aka `app/page.tsx`. Remove this from `RegionSelect.tsx` and put it in `app/page.tsx`.

It would be better to make your own API call to `api/hotels/region` within `LocationCombobox`.

In the future, put your custom components in `app/page.tsx` to have full access to all booking info.

Reply:
That's bad practice. Components should be abstracted and organized into files. This improves performance by allowing us to have less client components. It also improves development velocity by abstracting unimportant details and replacing them with a descriptive names. Props should be passed in as needed to expose information.

I didn't think about calling the API call directly, that might be the way forward. I was working in Region Select because that seemed to be the component that handled populating the region options.